bm-persona/Drawer/DrawerViewController.swift

```diff
@@ -8,55 +8,40 @@
 
 import UIKit
 
-enum DrawerState {
-    case hidden
-    case collapsed
-    case middle
-    case full
-}
-
+// General drawer with a gray bar at the top, can handle pan gesture to change position
 class DrawerViewController: UIViewController {
     var delegate: DrawerViewDelegate!
-    var state: DrawerState = .collapsed
-    var bInitialized: Bool = false
-    var heightOffset: CGFloat?
+    
+    // the current position of the drawer
+    var currState: DrawerState!
+    
+    // the last non-hidden position of the drawer (used to return drawer to correct position if it's currently hidden/covered)
+    var prevState: DrawerState!
+    var barView: UIView = UIView()
     
     override func viewDidLoad() {
         super.viewDidLoad()
         // Do any additional setup after loading the view.
-    }
-    
-    override func viewDidAppear(_ animated: Bool) {
-        if !bInitialized {
-            setupBackgroundView()
-            setupGestures()
-            bInitialized = true
-        }
+        // default to collapsed initial position
+        currState = .collapsed
+        prevState = .collapsed
+        
+        setupBackgroundView()
+        setupGestures()
     }
     
     func setupBackgroundView() {
-        let backgroundView = UIView()
-        backgroundView.backgroundColor = Color.modalBackground
+        view.backgroundColor = Color.modalBackground
+        view.layer.cornerRadius = 50
```

Comment:
Updated this to be 15. I think with that could we move the content in the drawer up a bit?

Reply:
updated and the content was so far down because of a constraint issue which is fixed. distance from the top is now much smaller